File: ReverseInteger.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<stdbool.h>
#include<math.h>
int reverse(int x)
{
    if (x == -2147483648)   return 0;
	bool flag = x < 0 ? 1 : 0;
    if (flag)   x = -x;
	char s[11];
	int pos = 0;
	while (x)
	{
		s[pos ++] = x % 10 + '0';
		x /= 10;
	}
	s[pos] = 0;
	int base = 1, begin = -1, out = 0;
	while (!s[++ begin]);
    int a[10] = {2, 1, 4, 7, 4, 8, 3, 6, 4, 7};
    if (pos - begin == 10)
        for (int i = 0; i < 10; ++ i)
        {
            if (s[i] - '0' > a[i])  return 0;
            if (s[i] - '0' < a[i])  break;
        }
	for (int i = pos - 1; i >= begin; -- i)
	{
		out += base * (s[i] - '0');
        if (pos == 10 && i == 0) break;
		base *= 10;
	}
    return flag ? -out : out;
}
int main()
{
	FILE *fin = fopen("oo.xx", "r");
	FILE *fout = fopen("xx.oo", "w");
	int n;
	fscanf(fin, "%d", &n);
	int out = reverse(n);
	printf("%d\n", out);
	fprintf(fout, "%d\n", out);
	return 0;
}

File: ReverseInteger.py
import functools as ft
from functools import partial as par
def F(*z):
    z = [*z]
    return [*ft.reduce(lambda x, y: map(y, x), [z[:1]] + z[1:])][0]
FF = lambda *z: [*ft.reduce(lambda x, y: map(y, x), z)]
fyx = lambda f, *x: lambda *y: f(*y, *x)
class Solution:
    def reverse(self, x: int) -> int:
        if x == -2147483648:
            return 0
        flag = 1 if x < 0 else 0
        x = -x if flag else x
        headFlag, s = 0, []
        upperLimit = [2, 1, 4, 7, 4, 8, 3, 6, 4, 7] 
        while x:
            if x % 10:
                s.append(x % 10)
                headFlag = 1
            elif headFlag:
                s.append(x % 10)
            x //= 10
        lens = len(s)
        if not len(s):
            return 0
        if lens == 10:
            for i in range(10):
                if s[i] > upperLimit[i]:
                    return 0
                if s[i] < upperLimit[i]:
                    break
        out, base = 0, 1
        s = s[::-1]
        for x in s[:-1]:
            out += x * base
            base *= 10
        out += s[-1] * base
        return -out if flag else out
        
fin = open("oo.xx", "r")
fout = open("xx.oo", "w")
n = int(fin.readline())
out = Solution().reverse(n)
print(out)
F(out, str, fout.write)

